refactor: remove references of ENABLE_HOME_PAGE_COURSE_API_V2

bra-i-am opened the pull request:
## Description

This PR primarily aims to eliminate references to ENABLE_HOME_PAGE_COURSE_API_V2: it intends to use that API, enabling the pagination and the course filters by default without needing the flag. Moreover, it also solves some issues related to the requests to the API that make the UI blink or request more times than required.

## Supporting information

https://github.com/eduNEXT/consulting-issues-mapping/issues/118

## Side effects
- The `Archived Courses Tab` has been removed as it is no longer necessary because it relied on the previously used attribute `archived_courses`, which is no longer in use. With the implementation of the courses API v2, courses are marked as archived when the `isActive` attribute of the retrieved courses is `true`, as reviewed through the new filters.

## Testing instructions

1. Mount this MFE on this specific branch in a Sumac/Nightly environment
2. Be sure you count with many courses (11 makes to appear the pagination component)
3. You should watch by default the search bar and filters without adding any env variable

    ![image](https://github.com/user-attachments/assets/a8226708-68f9-420d-b8ed-d32ed28d6def)

5. Open the inspect and go to the Network tab, every time you change the search bar input or change any filter, you should see only one endpoint call is made

openedx-webhooks commented:
Thanks for the pull request, @bra-i-am!


This repository is currently maintained by `@openedx/2u-tnl`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary><h4>:radio_button: Get product approval</h4></summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details><details>
<summary><h4>:radio_button: Provide context</h4></summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details><details>




<summary><h4>:radio_button: Get a green build</h4></summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>

<summary><h4>:radio_button: Update the status of your PR</h4></summary>

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.
</details><details>
<!-- comment:end_of_wip -->


---

<summary><h3>Where can I find more information? </h3></summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details><details>
<summary><h3>When can I expect my changes to be merged?</h3></summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

mariajgrimaldi commented:
@bradenmacdonald: Can you help us review this PR which is part of the https://github.com/openedx/public-engineering/issues/287 deprecation effort? Thank you!

bradenmacdonald commented:
> * The `Archived Courses Tab` has been removed as it is no longer necessary because it relied on the previously used attribute `archived_courses`, which is no longer in use. With the implementation of the courses API v2, courses are marked as archived when the `isActive` attribute of the retrieved courses is `true`, as reviewed through the new filters.
> * Taking into account the conditional from [courses-tab](https://github.com/openedx/frontend-app-authoring/blob/master/src/studio-home/tabs-section/courses-tab/index.tsx#L135),  should only be displayed when `isEnablePagination` is `false`. Since it cannot currently function this way, I assume this feature should be dropped. I wasn't able to make this work without hardcoding (I’m not sure if I'm doing something wrong or if the feature is not functioning). However, based on my review of the code, when a course is executed for re-run, it should transition to a [in_process](https://github.com/openedx/edx-platform/blob/2cb9aef33222d32fc09f7736878bcf39dd7b0923/cms/djangoapps/contentstore/views/course.py#L383) state, which I believe represents a background task, and this is indicated on the front end through the ProcessingCourses component. I’m unclear whether the element should be rendered or not with the new features (I didn't quite understand it), but currently, when the re-run is initiated, the user is redirected to a new route with its own loader button indicating the action being processed. Do you happen to know if this feature should be skipped or how I can get this to work for testing? Should we work this on another PR?

Hmm, I don't think I'll be able to answer these questions. Maybe @jristau1984 or @KristinAoki can answer?

KristinAoki commented:
I think the ProcessingCourses component work should be moved to another PR. I have only seen it one time when a course creation job was hung. I am unable to say if it should be skipped or not. If you want to see the feature, you can manually force a course be in-progress by hard-coding the status.

bra-i-am commented:
> I think the ProcessingCourses component work should be moved to another PR.

@KristinAoki, taking this into account, I removed the ProcessingCourses related changes from this merge request and left only the related with dropping ENABLE_HOME_PAGE_COURSE_API_V2

mphilbrick211 commented:
Hi @saadyousafarbi and @bra-i-am! Is this still in progress?

bradenmacdonald commented on src/studio-home/card-item/CardItem.test.tsx:
```diff
@@ -29,15 +29,17 @@ describe('<CardItem />', () => {
     render(<CardItem {...props} />);
     const courseTitleLink = screen.getByText(props.displayName);
     expect(courseTitleLink).toHaveAttribute('href', `${getConfig().STUDIO_BASE_URL}${props.url}`);
+    const dropDownMenu = screen.getByTestId('toggle-dropdown');
```
Please try to [avoid `getByTestId`](https://testing-library.com/docs/queries/about#priority). In fact, I can see there's an accessibility issue with this button - it doesn't have a `name` for non-visual users
![Screenshot](https://github.com/user-attachments/assets/4dc83470-166e-4cd8-beaa-a0a44538c9a3)

![Screenshot](https://github.com/user-attachments/assets/f24711e4-1d68-45e0-bdbf-8b0cf5a42ad1)

Please add an `aria-label` to the button then change this to something like `screen.getByRole('button', {name: /Course Actions/i})`

bradenmacdonald commented on src/studio-home/card-item/index.tsx:
```diff
@@ -92,48 +87,27 @@ const CardItem: React.FC<Props> = ({
+            as={IconButton}
+            iconAs={MoreHoriz}
+            variant="primary"
+            data-testid="toggle-dropdown"
```
As mentioned above, I think this needs an `aria-label`:
```suggestion
            aria-label={intl.formatMessage(messages. ... )}
```

See also https://github.com/openedx/paragon/issues/3616

bradenmacdonald commented on src/studio-home/tabs-section/courses-tab/index.test.tsx:
```diff
@@ -141,4 +131,17 @@ describe('<CoursesTab />', () => {
+    const props = { isLoading: false, coursesDataItems: [] };
+    const customStoreData = { studioHomeCoursesRequestParams: { isFiltered: true } };
+    const { store } = renderComponent(props, customStoreData);
+    const cleanFiltersButton = screen.queryByTestId('clean-filters');
```
```suggestion
    const cleanFiltersButton = screen.queryByRole('button', {name: /Clear filters/i});
```

bradenmacdonald commented on src/studio-home/tabs-section/courses-tab/index.tsx:
```diff
@@ -199,7 +184,7 @@ const CoursesTab: React.FC<Props> = ({
               {intl.formatMessage(messages.coursesTabCourseNotFoundAlertMessage)}
             </p>
-            <Button variant="primary" onClick={handleCleanFilters}>
+            <Button data-testid="clean-filters" variant="primary" onClick={handleCleanFilters}>
```
```suggestion
            <Button variant="primary" onClick={handleCleanFilters}>
```

bradenmacdonald reviewed (commented):
A couple minor requests. Also can you please rebase this? It seems a bit old at this point. Then I'll test and approve, and we can get it merge. Thanks!

bra-i-am commented:
@bradenmacdonald,

I apologize; it was just a bit outdated :sweat_smile:. I have already addressed your comments, and I believe this is ready.

Thank you so much for your review! :sparkles:
